### PR TITLE
[@types/next] Fix getInitialProps typings for the _error page

### PR DESCRIPTION
**Pull request by martpie**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Something weird is the test are not currently plassing on `master` (before my change), so I am not sure what I should do.

### CODE REVIEW

**Review by jthegedus — APPROVED** [review]

**Comment by resir014** [comment]
Seems like a similar error to #29890. I'm also not sure what to do with it either.

/cc @joaovieira @andy-ms

File: types/next/error.d.ts [hunk 1]
```diff
@@ -10,5 +10,5 @@ export interface DefaultErrorIProps {
 }
 
 export default class Error<P = {}> extends React.Component<P & DefaultErrorIProps> {
-    static getInitialProps(context: NextContext): DefaultErrorIProps;
+    static getInitialProps(context: NextContext): Promise<DefaultErrorIProps>;
```
Comment (joaovieira) [inline types/next/error.d.ts]: Can we change it to `DefaultErrorIProps | Promise<DefaultErrorIProps>`? The method in the exported Error class from Next.js returns an object (https://github.com/zeit/next.js/blob/canary/packages/next/pages/_error.js#L7)

Comment (martpie) [reply types/next/error.d.ts]: Well, this is actually interesting, I though it was a mistake from the typings, but it looks like a problem with my implementation, I used the `async` keyword on `getInitialProps` when I actually did not need it.

Do you think the proposal of having it as potentially a Promise is the good solution? Or should it always be NOT a Promise?

If the latter, then I guess we can close this PR.

Comment (joaovieira) [reply types/next/error.d.ts]: It's a weird one that happened several times in the past as well - and I still didn't completely wrap my head around it. The typings match the exported classes. When you extend classes the methods should keep the same signatures. But Next.js multiple return types.

We can keep the PR with my suggested change. Though we should think about it properly and holistically - this happens to all Next.js components (App, Document, Error).

**Comment by martpie** [comment]
Well my dear bot, I am going to rebase from origin/master, but I'm afraid there's a deeper problem going on here.

**Review by resir014 — APPROVED** [review]
The CI errors are a known issue with `typescript@next` and React. This should be good to go still.

**Comment by martpie** [comment]
Seems like test are now passing :)

**Pull request merged**